object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -338,44 +339,135 @@ private void handleAssignment(AssignmentNode node, Set<LocalVarWithTree> newDefs
     if (lhsElement.getKind().equals(ElementKind.FIELD) && typeFactory.hasMustCall(lhs.getTree())) {
       if (rhs instanceof LocalVariableNode && isVarInDefs(newDefs, (LocalVariableNode) rhs)) {
         if (typeFactory.getDeclAnnotation(lhsElement, Owning.class) != null) {
-          LocalVarWithTree latestAssignmentPair =
-              getAssignmentTreeOfVar(newDefs, (LocalVariableNode) rhs);
-          newDefs.remove(latestAssignmentPair);
+          Set<LocalVarWithTree> setContainsLatestAssignmentPair =
+              getSetContainsAssignmentTreeOfVar(newDefs, (LocalVariableNode) rhs);
+          newDefs.remove(setContainsLatestAssignmentPair);
         }
       }
     } else if (lhs instanceof LocalVariableNode
         && !isTryWithResourcesVariable((LocalVariableNode) lhs)) {
 
       // Reassignment to the lhs
       if (isVarInDefs(newDefs, (LocalVariableNode) lhs)) {
+        Set<LocalVarWithTree> setContainsLatestAssignmentPair =
+            getSetContainsAssignmentTreeOfVar(newDefs, (LocalVariableNode) lhs);
         LocalVarWithTree latestAssignmentPair =
             getAssignmentTreeOfVar(newDefs, (LocalVariableNode) lhs);
-        checkMustCall(
-            latestAssignmentPair,
-            typeFactory.getStoreBefore(node),
-            "variable overwritten by assignment " + node.getTree());
-        newDefs.remove(latestAssignmentPair);
+        if (setContainsLatestAssignmentPair.size() == 1) {
+          checkMustCall(
+              newDefs,
+              setContainsLatestAssignmentPair,
+              typeFactory.getStoreBefore(node),
+              "variable overwritten by assignment " + node.getTree());
+          newDefs.remove(setContainsLatestAssignmentPair);
+        } else {
+          setContainsLatestAssignmentPair.remove(latestAssignmentPair);
+        }
       }
 
       // If the rhs is an ObjectCreationNode, or a MethodInvocationNode, then it adds
       // the AssignmentNode to the newDefs.
       if ((rhs instanceof ObjectCreationNode)
           || (rhs instanceof MethodInvocationNode
               && !hasNotOwningReturnType((MethodInvocationNode) rhs))) {
-        newDefs.add(
-            new LocalVarWithTree(new LocalVariable((LocalVariableNode) lhs), node.getTree()));
+        if (typeFactory.hasMustCallChoice(rhs.getTree())) {
+          Node n = getMustCallChoiceParam(rhs, newDefs);
+          if (n != null && isVarInDefs(newDefs, (LocalVariableNode) n)) {
```

Comment:
How do we know that `n` will be a `LocalVariableNode`?

Reply:
Because getMustCallChoiceParam() returns LocalVariableNode if any exists, or returns null.